### PR TITLE
Add support for relative path option

### PR DESCRIPTION
**Pull request by bgalletbgallet**
Add the possibility to use --relative=true option in command line
This is useful if you run your test on a linux environment (docker, vm, server) and dev on a windows one.
If you want your coverage to work with your IDE, you need relative path instead of absolute ones.

Linked with pull request https://github.com/istanbuljs/istanbuljs/pull/492

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -428,6 +428,7 @@ class NYC {
       reports.create(_reporter, {
         skipEmpty: this.config.skipEmpty,
         skipFull: this.config.skipFull,
+        relative: this.config.relative === 'true',
```
Comment (coreyfarrell) [inline index.js]:
```suggestion
        cwd: this.cwd,
```

Comment (bgalletbgallet) [reply index.js]: Hi @coreyfarrell 
Shouldn't I use this.config.cwd to access the cwd option ?

Comment (coreyfarrell) [reply index.js]: No, please use `this.cwd`.  In some edge cases it's possible that `this.config.cwd` might be unset but this.cwd has the default `process.cwd()`.

https://github.com/istanbuljs/nyc/blob/master/index.js#L60

Comment (bgalletbgallet) [reply index.js]: Perfect I didn't see it was already set. I do it tonight
Thank you for your anwser

Comment (coreyfarrell) [reply index.js]: I've altered the suggested change per my latest comment on the istanbuljs PR.

**Comment by bgalletbgallet** [comment]
@coreyfarrell Everything should be ok, the issue with --cwd option is still present but this PR is not concerned anymore.
By default the lcov report is relative. A customer projectRoot option can be used to change the 'from' argument of the path.relative() function

**Comment by coreyfarrell** [comment]
@bcoe This adds a `projectRoot` option to nyc which could be set differently than `cwd`.  Do you think this is useful?  I'm concerned this could cause confusion for users.

Ref #1234 - the config loading is complicated and potentially needs some tweaks before we do a final release of nyc 15.

**Comment by bcoe** [comment]
@coreyfarrell this would be extremely useful to me, I'm currently floating a patch for the Node.js nightly builds, with almost this functionality (so that merging works):

https://codecov.io/gh/nodejs/node/src/64ee64d3afc837350ee39058cef9179e92f3554c/lib/fs.js

**Comment by coreyfarrell** [comment]
@bgalletbgallet could you explain to me why we need a separate option for `projectRoot` compared to `cwd`?  I'm very concerned this will cause confusion to end users, and interested in knowing about a scenario where the value of `cwd` will would be unacceptable as the base of reporting.  Keep in mind that by default `cwd` will be the directory containing the package.json which applies to `process.cwd()`.  I'm afraid to add complexity without understanding the justification.

A minimal reproduction which installs `git://github.com/bgalletbgallet/nyc#master` for the nyc dependency showing that it is necessary to set `projectRoot` to get functional results would be great.

**Comment by bgalletbgallet** [comment]
@coreyfarrell I don't know, I was about to delete it and @bcoe told that it was usefull to him. I don't see any case where you need a different projectRoot than the default cwd. If it's ok for @bcoe, we will delete it

File: index.js [hunk 2]
```diff
@@ -440,6 +441,7 @@ class NYC {
       reports.create(_reporter, {
         skipEmpty: this.config.skipEmpty,
         skipFull: this.config.skipFull,
+        projectRoot: this.projectRoot,
```
Comment (coreyfarrell) [inline index.js]:
```suggestion
        projectRoot: this.cwd,
```

File: index.js [hunk 3]
```diff
@@ -57,6 +57,7 @@ class NYC {
     this._showProcessTree = config.showProcessTree || false
     this._eagerInstantiation = config.eager || false
     this.cwd = config.cwd || process.cwd()
+    this.projectRoot = config.projectRoot || this.cwd
```
Comment (coreyfarrell) [inline index.js]: Let's please delete this line.

**Pull request merged**